api/filter/http_connection_manager.proto

```diff
@@ -99,55 +99,54 @@ message HeaderSelection {
 }
 
-message AccessLog {
-  // Destination location the access log is written to.
-  //
-  // Envoy supports logging to both RPC endpoints and local disk files.
-  // Behavior is set by choosing which of these is used:
-  oneof destination {
-    // A path to a local file
-    string path = 1;
-    // The name of an upstream cluster that hosts RPC endpoints for the
-    // AccessLog sink. The cluster must provide an endpoint capable of
-    // receiving the AccessLog API. The exact flavor of API spoken will
-    // depend on the statically registered log interface module specified
-    // in Envoy's startup configuration.
-    string cluster = 5;
-  }
+// Custom configuration for an AccessLog that writes log entries directly to a file.
+// Configures the built-in "envoy.file_access_log" AccessLog.
+message FileAccessLog {
+  // A path to a local file to which to write the access log entries.
+  string path = 1;
+
+  // [V2-API-DIFF] In addition to the existing format specifiers, the following
+  // additional specifiers will be available:
+  // %TLS_SNI_HOSTNAME%: SNI from handshake.
+  // %TLS_VERSION%: Possible values include: “TLSv1”, “TLSv1.1”,
+  //     “TLSv1.2”, “TLSv1.3”.
+  // %TLS_CIPHER_SUITE%: Cipher suite negotiated during the TLS handshake. The
+  //     value is four hex digits defined by the IANA TLS Cipher Suite Registry,
+  //     e.g. “009C” for TLS_RSA_WITH_AES_128_GCM_SHA256.
+  // TODO(htuch): Document how envoy.http_connection_manager.access_log values
+  // can be accessed in the format specifier.
+  string format = 2;
+}
 
-  // Content included in log entries.
-  //
-  // When logging to a file this is a format string which specifies what is
-  // included.
-  //
-  // When logging to an RPC endpoint all of the named fields in accesslog.proto
-  // are populated by default if the relevant data is available for the logged
-  // request. Additional headers desired for logging can be specified in the
-  // HeaderSelection.
-  oneof content {
-    // Access log format for file logging.
-    //
-    // Envoy supports custom access log formats as well as a
-    // default format.
-    //
-    // [V2-API-DIFF] In addition to the existing format specifiers, the following
-    // additional specifiers will be available:
-    // %TLS_SNI_HOSTNAME%: SNI from handshake.
-    // %TLS_VERSION%: Possible values include: “TLSv1”, “TLSv1.1”,
-    //     “TLSv1.2”, “TLSv1.3”.
-    // %TLS_CIPHER_SUITE%: Cipher suite negotiated during the TLS handshake. The
-    //     value is four hex digits defined by the IANA TLS Cipher Suite Registry,
-    //     e.g. “009C” for TLS_RSA_WITH_AES_128_GCM_SHA256.
-    // TODO(htuch): Document how envoy.http_connection_manager.access_log values
-    // can be accessed in the format specifier.
-    string format = 2;
-
-    // Extra headers to include when RPC logging.
-    HeaderSelection headers = 4;
-  }
+// Custom configuration for an AccessLog that sends logs to an RPC endpoint.
```

Comment:
You give a `envoy.foo` name above for `FileAccessLog`, but not one for here.

Reply:
I'm assuming we won't have a valid RPC access log implementation in Envoy (as opposed to the file one, which already exists) when we begin consuming this and using statically registered access logs.  So I thought it would be confusing to give a string that would ultimately throw an error when specified as a name.

Reply:
If we don't have a reference RPC implementation, then maybe we shouldn't even have this in the API, it should just be private to our branch (same goes with `accesslog.proto`).

Reply:
That seems fine to me.  I'll remove them, and we can add back later when that exists in Envoy.